spyt-package/src/main/python/spyt/conf.py

```diff
@@ -26,6 +18,71 @@
 logger = logging.getLogger(__name__)
 
 
+class SpytDistributions:
+    def __init__(self, client: YtClient, yt_root: str):
+        self.client = client
+        self.yt_root = YPath(yt_root)
+        self.conf_base_path = self.yt_root.join("conf")
+        self.global_conf = client.get(self.conf_base_path.join("global"))
+        self.distrib_base_path = self.yt_root.join("distrib")
+        self.spyt_base_path = self.yt_root.join("spyt")
+
+    def read_remote_conf(self, cluster_version):
+        version_conf = self.client.get(self._get_version_conf_path(cluster_version))
+        version_conf["cluster_version"] = cluster_version
+        return update_inplace(self.global_conf, version_conf)  # TODO(alex-shishkin): Might cause undefined behaviour
+
+    def latest_ytserver_proxy_path(self, cluster_version):
+        if cluster_version:
+            return None
+        global_conf = self.global_conf
+        symlink_path = global_conf.get("ytserver_proxy_path")
+        if symlink_path is None:
+            return None
+        return get("{}&/@target_path".format(symlink_path), client=self.client)
+
+    def validate_cluster_version(self, spark_cluster_version):
+        if not exists(self._get_version_conf_path(spark_cluster_version), client=self.client):
+            raise RuntimeError("Unknown SPYT cluster version: {}. Available release versions are: {}".format(
+                spark_cluster_version, self.get_available_cluster_versions()
+            ))
+        spyt_minor_version = SpytVersion(SELF_VERSION).get_minor()
+        cluster_minor_version = SpytVersion(spark_cluster_version).get_minor()
+        if spyt_minor_version < cluster_minor_version:
+            logger.warning("You required SPYT version {} which is older than your local ytsaurus-spyt version {}."
+                           "Please update your local ytsaurus-spyt".format(spark_cluster_version, SELF_VERSION))
+
+    def get_available_cluster_versions(self):
+        subdirs = yt_list(self.conf_base_path.join(RELEASES_SUBDIR), client=self.client)
```

Comment:
`subdirs` can be obtained by calling `self.get_available_spyt_versions()` so it will remove code duplication

Reply:
There are two different paths used there, so this method cannot be used.

Reply:
Oh, I see, `conf_base_path` and `spyt_base_path`, you're right